host/Cargo.toml

```diff
@@ -10,9 +10,12 @@ workspace = true
 
 [dependencies]
 arrow.workspace = true
+datafusion.workspace = true
```

Comment:
I kinda hate that we now need the entire DF crate ecosystem just because of the `SessionContext` living in the "core" crate (it's not even a "core" at this point). CC @alamb 

BTW: for new dependencies, can you try `default_features = false` so that we don't pull in all kinds of nonesense?